src/components/Proposal/ActionButton.tsx

```diff
@@ -217,4 +243,33 @@ class ActionButton extends React.Component<IProps, IState> {
   }
 }
 
-export default connect(mapStateToProps, mapDispatchToProps)(ActionButton);
+const SubscribedActionButton = withSubscription({
```

Comment:
@jellegerbrandy  @tibetsprague Should the gen and externalToken amounts be computed in `ProposalData`?

Reply:
To answer my own question:  I don't think so because `ActionButton` is used outside of `ProposalData`, in `RedemptionsMenu`

Reply:
In fact I think an argument could be made that the computation of ethBalance should be moved from ProposalData and RedemptionsMenu into the ActionButton.